src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -281,9 +282,78 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto actionPaletteItem{ filteredCommand.Item().try_as<winrt::TerminalApp::ActionPaletteItem>() })
             {
-                PreviewAction.raise(*this, actionPaletteItem.Command());
+                const auto& cmd = actionPaletteItem.Command();
+                PreviewAction.raise(*this, cmd);
+
+                const auto description{ cmd.Description() };
+
+                if (const auto& selected{ SelectedItem() })
+                {
+                    selected.SetValue(Automation::AutomationProperties::FullDescriptionProperty(), winrt::box_value(description));
+                }
+
+                if (!description.empty())
+                {
+                    _openTooltip(cmd);
+                }
+                else
+                {
+                    // If there's no description, then just close the tooltip.
+                    _descriptionsView().Visibility(Visibility::Collapsed);
+                    _descriptionsBackdrop().Visibility(Visibility::Collapsed);
+                    _recalculateTopMargin();
+                }
+            }
+        }
+    }
+
+    void SuggestionsControl::_openTooltip(Command cmd)
+    {
+        const auto description{ cmd.Description() };
+        if (description.empty())
+        {
+            return;
+        }
+
+        // Build the contents of the "tooltip" based on the description
+        //
+        // First, the title. This is just the name of the command.
+        _descriptionTitle().Inlines().Clear();
+        Documents::Run titleRun;
+        titleRun.Text(cmd.Name());
+        _descriptionTitle().Inlines().Append(titleRun);
+
+        // Now fill up the "subtitle" part of the "tooltip" with the actual
+        // description itself.
+        const auto& inlines{ _descriptionComment().Inlines() };
```

Comment:
```suggestion
        const auto inlines{ _descriptionComment().Inlines() };
```

this is a bad habit, we should not perpetuate it